File: Leetcode/387. First Unique Character in a String/solution.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        counts = {}

        for char in s:
            counts[char] = counts.get(char, 0) + 1

        for i, char in enumerate(s):
            if counts.get(char) == 1:
                return i

        return -1

File: Leetcode/387. First Unique Character in a String/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn first_uniq_char(s: String) -> i32 {
        let mut count = HashMap::new();

        for chr in s.chars() {
            *count.entry(chr).or_insert(0) += 1;
        }

        for (i, chr) in s.chars().enumerate() {
            if count.get(&chr) == Some(&1) {
                return i as i32;
            }
        }

        -1
    }
}
